31). Add topic files 91 to 94 (#18299)
dotnet/docs: docs/standard/linq/concepts-terminology-functional-transformation.md

> [!IMPORTANT]
> In the rest of this tutorial, the term "pure function" is used in a general sense to indicate a programming approach, and not a specific language feature.
>
> Note that pure functions must be implemented as methods in C#, and as functions in Visual Basic.
>
> You should not confuse pure functions with pure virtual methods in C++. The latter indicates that the containing class is abstract and that no method body is supplied.

### Functional programming

*Functional programming* is a programming approach that directly supports pure functional transformation.

Historically, general-purpose functional programming languages, such as ML, Scheme, Haskell, and F#, have been primarily of interest to the academic community. Although it has always been possible to write pure functional transformations in C# and Visual Basic, the difficulty of doing so has not made it an attractive option to most programmers. In recent versions of these languages, however, new language constructs such as lambda expressions and type inference make functional programming much easier and more productive.

> For more information about functional programming, see [Functional programming vs. imperative programming](functional-programming-vs-imperative-programming.md).

Reviewer: Let's shorten this file to avoid dupe words

```suggestion
For more information about functional programming, see [Functional programming vs. imperative programming](functional-vs-imperative-programming.md).
```

> #### Domain-specific FP languages

Reviewer: I imagine FP here stands for functional programming? Should we add the acronym definition to the previous heading?

Author: I removed the FP in favor of functional programming

> Although general functional programming languages have not been widely adopted, some domain-specific functional programming languages have had better success. For example, Cascading Style Sheets (CSS) are used to determine the look and feel of many Web pages, and Extensible Stylesheet Language Transformations (XSLT) style sheets are used extensively in XML data manipulation. For more information about XSLT, see [XSLT Transformations](../../standard/data/xml/xslt-transformations.md).

Reviewer: ```suggestion
Although general functional programming languages haven't been widely adopted, some domain-specific functional programming languages have had better success. For example, Cascading Style Sheets (CSS) are used to determine the look and feel of many web pages, and Extensible Stylesheet Language Transformations (XSLT) style sheets are used extensively in XML data manipulation. For more information about XSLT, see [XSLT Transformations](../data/xml/xslt-transformations.md).
```

## Terminology

> The following table defines some terms related to functional transformations.

Reviewer: This is not really a table. Maybe style the list a bit different too either as an unbulleted list or bold terms?

```suggestion
The following list defines some terms related to functional transformations.
```

Author: If I had found a quick easy way to restyle I would have done it. I did look at it a bit, don't remember what the difficulties were.

deferred execution and lazy evaluation \
The delaying of evaluation of an expression until its resolved value is actually required. Deferred execution is supported in collections.

For more C# information, see [Introduction to LINQ Queries (C#)](../../csharp/programming-guide/concepts/linq/introduction-to-linq-queries.md) and [Deferred Execution and Lazy Evaluation in LINQ to XML (C#)](../../csharp/programming-guide/concepts/linq/deferred-execution-and-lazy-evaluation-in-linq-to-xml.md).

For more Visual Basic information, see [Basic Query Operations (Visual Basic)](../../visual-basic/programming-guide/concepts/linq/basic-query-operations.md) and [Deferred Execution and Lazy Evaluation in LINQ to XML (Visual Basic)](../../visual-basic/programming-guide/concepts/linq/deferred-execution-and-lazy-evaluation-in-linq-to-xml.md).

These language features will be used in code samples throughout this section.

## See also

- [Introduction to pure functional transformations](pure-functional-transformations.md)

> - [Functional programming vs. imperative programming](functional-programming-vs-imperative-programming.md)

Reviewer: ```suggestion
- [Functional programming vs. imperative programming](functional-vs-imperative-programming.md)
```
